tfjs-backend-webgpu/src/shader_preprocessor_wgsl.ts

```diff
@@ -245,48 +245,48 @@ const SHADER_PREFIX = `
   }
 
   // Checks whether coordinates lie within the bounds of the shape.
-  fn coordsInBounds4D(coord : vec4<u32>, shape : vec4<u32>) -> bool {
+  fn coordsInBounds4D(coord : vec4<i32>, shape : vec4<i32>) -> bool {
     return all(coord < shape);
   }
 
-  fn coordsInBounds3D(coord : vec3<u32>, shape : vec3<u32>) -> bool {
+  fn coordsInBounds3D(coord : vec3<i32>, shape : vec3<i32>) -> bool {
     return all(coord < shape);
   }
 
-  fn coordsInBounds2D(coord : vec2<u32>, shape : vec2<u32>) -> bool {
+  fn coordsInBounds2D(coord : vec2<i32>, shape : vec2<i32>) -> bool {
     return all(coord < shape);
   }
   `;
 const SAMPLING_SNIPPETS = `
-  fn getFlatIndex1D(coord : u32, shape : u32) -> u32 {
+  fn getFlatIndex1D(coord : i32, shape : i32) -> i32 {
     return coord;
   }
 
-  fn getFlatIndex2D(coords : vec2<u32>, shape : vec2<u32>) -> u32 {
-    return u32(dot(vec2<f32>(coords), vec2<f32>(f32(shape.y), 1.0)));
+  fn getFlatIndex2D(coords : vec2<i32>, shape : vec2<i32>) -> i32 {
+    return i32(dot(vec2<f32>(coords), vec2<f32>(f32(shape.y), 1.0)));
   }
 
-  fn getFlatIndex3D(coords : vec3<u32>, shape : vec3<u32>) -> u32 {
-    return u32(dot(vec3<f32>(coords), vec3<f32>(f32(shape.y) * f32(shape.z), f32(shape.z), 1.0)));
+  fn getFlatIndex3D(coords : vec3<i32>, shape : vec3<i32>) -> i32 {
+    return i32(dot(vec3<f32>(coords), vec3<f32>(f32(shape.y) * f32(shape.z), f32(shape.z), 1.0)));
   }
 
-  fn getFlatIndex4D(coords : vec4<u32>, shape : vec4<u32>) -> u32 {
-    return u32(dot(vec4<f32>(coords), vec4<f32>(
+  fn getFlatIndex4D(coords : vec4<i32>, shape : vec4<i32>) -> i32 {
+    return i32(dot(vec4<f32>(coords), vec4<f32>(
         f32(shape.y) * f32(shape.z) * f32(shape.w), f32(shape.z) * f32(shape.w), f32(shape.w), 1.0)));
   }
 
   // Only used when the y/z dimension of workgroup size is 1.
-  fn getGlobalIndex(globalId : vec3<u32>, localId : vec3<u32>) -> u32 {
-    if (uniforms.dispatchSize.y == 1u && uniforms.dispatchSize.z == 1u) {
+  fn getGlobalIndex(globalId : vec3<i32>, localId : vec3<i32>) -> i32 {
```

Comment:
How about changing it like below. And if it's the only place that `dispatchSize` is used, you can keep it as `u32` for convenience.
And `numWorkGroups` sounds like a better name compared with `dispatchSize`.
```
fn getGlobalIndex(globalId : vec3<u32>, localId : vec3<u32>) -> i32
{
   XXX
   XXX
   return i32(XXX);
}
```